clang/lib/Basic/DarwinSDKInfo.cpp

```diff
@@ -62,6 +63,28 @@ DarwinSDKInfo::RelatedTargetVersionMapping::parseJSON(
       Min, Max, MinValue, MaximumDeploymentTarget, std::move(Mapping));
 }
 
+static llvm::Triple::OSType parseOS(const llvm::json::Object &Obj) {
+  // The CanonicalName is the Xcode platform, a version, and an optional suffix,
+  // e.g. macosx16.0.
```

Comment:
The example doesn't have a suffix, as far as I can tell. Up to you if you want to make it more representative.

Reply:
The suffix isn't commonly used so there isn't a good example for that. But the format does technically support a suffix after the version. Maybe we should just say that the CanonicalName is the Xcode platform followed by a version and leave it ambiguous that anything comes after that since all we're doing is parsing out the platform anyway?

Reply:
Went with that.